WIP - helper examples

makella opened the pull request:
working on some basic examples of how to use the helper methods.

- [x] Part 1: 
  - Description of each method and defaults
  - Examples of each one over different geometry and themes
  - Note about types of source layers
  
- [x] Part 2:  
  - Customization parameters for each helper
  - Mulit-layer example

- [x] Basemaps and viewport

makella commented:
Ok! Here are helper method Part 1 and 2 example notebooks. Tagging @Jesus89 and @VictorVelarde  for review since @elenatorro I know you are getting the workshop ready (but definitely want your feedback too)! Related to that, feel free to use any of the maps as screenshots for your helper methods section. 

I know there are so many more examples to make and even ways to customize what I have here further (my mind is turning the wheels)... in the interest of keeping the focus on helper methods, I didn't go too crazy :).  

@jorgesancha I will look at adding the multi-layer example tomorrow (I have an idea). Also, once the review is complete, I will see about a screencast and also please feel free to use these for your meeting with Virginia.

Jesus89 commented:
This is awesome mamata :heart: 

Nice to see real world examples using the new API :smile: 

Note1: I'm not sure about showing the `_inspect` method since this is something we use internally.
Note2: You are using 'table_name' for the source, but it actually works with any of the sources available (table_name, sql_query, geojson, etc)
Note3: You need to import the basemaps for the part-2 notebook (from cartoframes.viz import basemaps)

makella commented:
@Jesus89 yay!! thank you for your eyes on this!!

> Note1: I'm not sure about showing the _inspect method since this is something we use internally.

oops forgot to delete! was an artifact from the original notebook i copied. thanks for pointing out!
> Note2: You are using 'table_name' for the source, but it actually works with any of the sources available (table_name, sql_query, geojson, etc)

good point!! let me think about how to incorporate this
> Note3: You need to import the basemaps for the part-2 notebook (from cartoframes.viz import basemaps)

ahhhh yessss... i had it in there but removed since the basemaps were drawing. will add that section back.

VictorVelarde commented:
I'm not going to be able to give you feedback today nor tomorrow, so feel free to merge into develop if you wish.

makella commented:
@VictorVelarde no worries!! i haven't had a chance to do @Jesus89 comments yet. but I'll ask him to review one more time then can merge.

makella commented:
- [x] Note1
- [ ] Note2
- [x] Note3

andy-esch commented on examples/helper-methods-part-1.ipynb:
```diff
@@ -0,0 +1,10676 @@
+       "  }\n",
+       "}\n",
+       "\n",
+       "const responsive = doc
```
How about passing `palette='[red, blue]'` (or whatever colors are usually used for GOP and DEM) instead of the default here?

makella replied:
@andy-esch as mentioned, that (and more!) come in part 2!

makella commented:
@Jesus89 fixed all of your notes! Thanks for the review!! If you want, feel free to look over the Multi-layer section in Part 2 and the description of sources in the opening paragraphs of Part 1.

@jorgesancha Part 2 includes multi-layer examples! (Great idea to do that!)

Jesus89 commented:
Hey, @makella it looks nice!

The content looks good to me, but I have an issue loading each notebook because it contains lots of maps. Maybe we could split the examples in different notebooks to improve the performance and the experience of the readers.

I have seen that there is a weird file `examples/.ipynb` to the PR. You can remove it and commit it to the PR.

makella commented:
@Jesus89 thanks! deleted that orphan file... in terms of performance and splitting content... let me look into that... i guess i can have a few different notebooks but will break each one's "flow". ugh. would like to move on from this but will see what i can do

Jesus89 commented:
OK. No problem, this is just a suggestion. Keeping 2 notebooks is fine too

makella commented:
@Jesus89  totally but a really good point. I was thinking some of the viewport/basemap stuff could be moved to another notebook as well as the multi-layer example... let's see!

makella commented:
@Jesus89 ok, removed some maps from part 1 (i had multiple maps to demonstrate the same helper) also removed some extra stuff from part 2 and made the viewport/basemap customization a separate example.

thanks!

makella commented:
merged!
